elyra/cli/pipeline_app.py

```diff
@@ -0,0 +1,165 @@
+    pipeline_name = os.path.splitext(os.path.basename(pipeline_abs_path))[0]
+
+    if not os.path.exists(pipeline_abs_path):
+        raise click.ClickException(f"Pipeline file not found: '{pipeline_abs_path}'\n")
+
+    with open(pipeline_abs_path) as f:
+        try:
+            pipeline_definition = json.load(f)
+        except ValueError as ve:
+            raise click.ClickException(f"Pipeline file is invalid: \n {ve}")
+
+    for pipeline in pipeline_definition["pipelines"]:
+        for node in pipeline["nodes"]:
+            if 'filename' in node["app_data"]:
+                abs_path = os.path.join(pipeline_dir, node["app_data"]["filename"])
+                node["app_data"]["filename"] = abs_path
+
+    # NOTE: The frontend just set the info for first pipeline, but shouldn't it
+    # search for the primary pipeline and set that?
+    # Setting `pipeline_definition["pipelines"][0]` for consistency.
+    pipeline_definition["pipelines"][0]["app_data"]["name"] = pipeline_name
+    pipeline_definition["pipelines"][0]["app_data"]["runtime"] = runtime
+    pipeline_definition["pipelines"][0]["app_data"]["runtime-config"] = runtime_config
+
+    return pipeline_definition
+
+
+def _execute_pipeline(pipeline_definition):
+    try:
+        with yaspin(text="Processing pipeline ..."):
+            # parse pipeline
+            pipeline_object = PipelineParser().parse(pipeline_definition)
+            # process pipeline
+            asyncio.get_event_loop().run_until_complete(PipelineProcessorManager.instance().process(pipeline_object))
+    except ValueError as ve:
+        raise click.ClickException(f'Error parsing pipeline: \n {ve}')
+    except RuntimeError as re:
+        raise click.ClickException(f'Error processing pipeline: \n {re} \n {re.__cause__}')
+
+
+def print_banner(title):
+    click.echo(Fore.CYAN + "────────────────────────────────────────────────────────────────" + Style.RESET_ALL)
+    click.echo(Fore.CYAN + "  {}".format(title) + Style.RESET_ALL)
+    click.echo(Fore.CYAN + "────────────────────────────────────────────────────────────────" + Style.RESET_ALL)
+    click.echo()
+
+
+def print_info(title, info_list):
+    click.echo(Fore.CYAN + "❯ {}".format(title) + Style.RESET_ALL)
+    for info_item in info_list:
+        if isinstance(info_item, str):
+            click.echo("  {}".format(info_item))
+        else:
+            click.echo("  {}: {}".format(info_item[0], info_item[1]))
+    click.echo()
+
+
+def print_version():
+    print_info("Version", ["elyra {}".format(__version__)])
+
+
+@click.group()
+@click.version_option(__version__, message='v%(version)s')
+def pipeline():
+    """
+    Run Elyra pipelines in your local environment or submit them to an external service,
+    such as Kubeflow Pipelines or Apache Airflow.
+
+    Find more information at: https://elyra.readthedocs.io/en/latest/
+    """
+    pass
+
+
+@click.command()
+@click.argument('pipeline_path')
+@click.option('--runtime',
```

Comment:
We talked about inferring runtime from `runtime-config`.  Since this would essentially be an API change if we did this _after_ initially supporting `--runtime`, and since its trivial code to infer the runtime from the config, and since you need to validate that the runtime-config exists anyway, I think we should remove `--runtime` in the first draft.

Reply:
Sounds good to me